### PR TITLE
Add support for deploying new blocking triggers

### PR DESCRIPTION
**Pull request by blidd-google**
GCIP now supports two new event types `beforeSendEmail` and `beforeSendSms`. This PR adds support for deploying the new auth blocking function types that are triggered on `beforeSendEmail` and `beforeSendSms` events.

### CODE REVIEW

File: src/deploy/functions/services/auth.ts [hunk 1]
```diff
@@ -82,13 +85,24 @@ export class AuthBlockingService implements Service {
...
+    } else if (endpoint.blockingTrigger.eventType === events.v1.BEFORE_SEND_EMAIL_EVENT) {
+      newBlockingConfig.triggers = {
+        ...newBlockingConfig.triggers,
+        beforeSendEmail: {
+          functionUri: endpoint.uri!,
+        },
+      };
+    } else {
+      throw new FirebaseError(
+        `Received invalid blocking trigger event type ${endpoint.blockingTrigger.eventType}`
```
Comment (Xiaoshouzi-gh) [inline src/deploy/functions/services/auth.ts:104]: quick question - seems like the error msg will be thrown to customers using firebase CLI, would new error codes and msg needs some sort of approval?

**Review by Xiaoshouzi-gh — APPROVED** [review]
LGTM

File: src/deploy/functions/services/auth.ts (hunk 1 above)
Comment (blidd-google) [reply src/deploy/functions/services/auth.ts:104]: The error message will be wrapped in an error object that will have a code related to a failed deployment.

**Review by Xiaoshouzi-gh — APPROVED** [review]

**Review by joehan — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Add support for deploying beforeEmailSent blocking function. (#6384)
```
Comment (joehan) [inline CHANGELOG.md:1]:
```suggestion
- Added support for deploying `beforeEmailSent` blocking functions. (#6384)
```

**Pull request merged**